src/cos_integration.py

```diff
@@ -100,45 +100,25 @@ def get_metrics_endpoints(self, node_name: str, token: str, cluster_name: str) -
             "replacement": node_name,
         }
 
+        components = [
+            ("kube-proxy", "/metrics", "http", "localhost:10249"),
+            ("apiserver", "/metrics", "https", "localhost:6443"),
+            ("kube-scheduler", "/metrics", "https", "localhost:10259"),
+            ("kube-controller-manager", "/metrics", "https", "localhost:10257"),
+        ]
```

Comment:
doesn't this just beg for a type :D

```python
from dataclasses import dataclass

class Component(dataclass):
    name: str
    path: str
    schema: str
    host: str
    instance_relabel: List[???]
```

Reply:
Let's rollback to the JobConfig definition, adding a typing would introduce another dataclass similar to the JobConfig 😄